### PR TITLE
packages: add amazon-ecr-credential-helper

### PR DESCRIPTION
**Pull request by leb4r**
## what

* Adds `amazon-ecr-credential-helper`  

## why

*  Use AWS credentials to perform actions against ECR

## references

* https://github.com/awslabs/amazon-ecr-credential-helper/releases

### CODE REVIEW

**Comment by nitrocode** [comment]
Added references. Looks good to me. Maybe get one more approval though.

cc: @Nuru @aknysh

**Review by nitrocode — DISMISSED** [review]

**Comment by nitrocode** [comment]
Hmmm actually on second thought, it looks like the docs weren't updated

**Review by nitrocode — APPROVED** [review]
Added references. Looks good to me. Maybe get one more approval though.

cc: @Nuru @aknysh

File: vendor/amazon-ecr-credential-helper/Makefile [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+export CHECK_COMMAND_ENV ?= HOME=./
+export PACKAGE_NAME ?= docker-credential-ecr-login
+
+include ../../tasks/Makefile.vendor_includes
+
+install:
+	$(call download_binary)
+
+test:
+	HOME=/tmp $(PACKAGE_EXE) version
```
Comment (Nuru) [inline vendor/amazon-ecr-credential-helper/Makefile:15]: Doesn't work that way, plus we check the version number if possible
```suggestion
	HOME=/tmp $(PACKAGE_EXE) -v | grep -F $(PACKAGE_VERSION)
```

**Review by Nuru — CHANGES_REQUESTED** [review]

**Comment by Nuru** [comment]
Remember to `make readme` after any changes. I prefer to do it under Docker:

```
make docker/build/deb/shell
make readme
exit
```

File: vendor/amazon-ecr-credential-helper/Makefile [hunk 2]
```diff
@@ -0,0 +1,15 @@
+# Package details
+export VENDOR ?= awslabs
+export APK_BUILD_TEMPLATE ?= APKBUILD.github-binary
+export DOWNLOAD_URL ?= https://amazon-ecr-credential-helper-releases.s3.us-east-2.amazonaws.com/$(PACKAGE_VERSION)/$(HOST_OS)-$(ARCH)/docker-credential-ecr-login
+export CHECK_COMMAND_ARGUMENTS ?= version
+export CHECK_COMMAND_ENV ?= HOME=./
+export PACKAGE_NAME ?= docker-credential-ecr-login
```
[the remark this answers is not in the document]
Comment (leb4r) [reply vendor/amazon-ecr-credential-helper/Makefile:7]: @Nuru I believe this is consistent elsewhere I've used it (see their [readme](https://github.com/awslabs/amazon-ecr-credential-helper#prerequisites)), where the package is named `amazon-ecr-credential-helper` but it installs `docker-credential-ecr-login` binary. 

Docker uses the prefix `docker-credential-*` as a means to search out the credential helpers on the system. 

Is there a better way to get the same results?

[the remark this answers is not in the document]
Comment (Nuru) [reply vendor/amazon-ecr-credential-helper/Makefile:7]: The `PACKAGE_NAME` is the name we are giving to the package in the package repo, e.g. `apt-get install $PACKAGE_NAME`.

[the remark this answers is not in the document]
Comment (Nuru) [reply vendor/amazon-ecr-credential-helper/Makefile:7]: The installed file will still be `$PACKAGE_EXE`, which is `docker-credential-ecr-login`

**Review by Nuru — APPROVED** [review]

**Pull request merged**